src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1474,6 +1476,47 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        //Returns true if a given DirectoryInfo path is a recursive sym link.
+        private bool IsRecursiveSymLink(DirectoryInfo path)
+        {
+            string readSymLink = "";
+
+            if (Platform.IsWindows)
+            {
+                PSObject obj = PSObject.AsPSObject(path);
+
+                try
+                {
+                    readSymLink = InternalSymbolicLinkLinkCodeMethods.GetTarget(obj).First().ToString();
+                }
+                catch
+                {
+                    readSymLink = null;
+                }
+
+                //if the read symlink is not null or empty, and what it points to is the parent directory then it is recursive
+                if (!String.IsNullOrEmpty(readSymLink) && path.Parent.FullName.Contains(readSymLink))
```

Comment:
> path.Parent.FullName.Contains(readSymLink))

I think `path.FullName.StartsWith` is a better check. Considering the following case:
```
path = /tmp/home/pi/foo
path.Parent.FullName = /tmp/home/pi
readSymLink = /home/pi
```
in this case, path is pointing to its parent recursively, but this method would return true.